### PR TITLE
[Docs] stoploss documentation improvements

### LINKED ISSUE #2428: `trailing_stop_positive` is a misleading name
### Suggestion to change this name or update the documentation

I had a strategy with the following configuration:
```    
    stoploss = -0.05
    trailing_stop = True
    trailing_stop_positive = 0.005
    trailing_stop_positive_offset = 0.0051
    trailing_only_offset_is_reached = False
```
The `trailing_stop_positive` and `trailing_stop_positive_offset` where just there for debugging to easily enable and disable them *I thought*.

I assumed that if `trailing_only_offset_is_reached` is `true` it used those 2 values `trailing_stop_positive` and `trailing_stop_positive_offset`

The docs say this about this option:
```If "trailing_only_offset_is_reached": true then the trailing stoploss is only activated once the offset is reached. Until then, the stoploss remains at the configuredstoploss.```

However what I didn't know if `trailing_only_offset_is_reached` is set to `false` the `trailing_stop_positive` value is always used and not the `stop_loss` value. This makes this name very misleading, since it's called 'positive' I thought it was only used when `trailing_only_offset_is_reached` is set to true. Apparently this is not the case.

It's a very small 'issue' but it might save someone in the future a lot of debugging time.

### PR DESCRIPTION
**Pull request by xmatthias**
## Summary
Improve the stoploss documentation 

closes #2428 

## Quick changelog

- by simplifying the documentation by not repeating that everything can be set in configuration or strategy
- changing code-samples to python (all user samples i've seen so far use stoploss in the strategy, not from the configuration) - so i assume this is the primary place to configure this
- split `trailing_only_offset_is_reached` from `trailing_stop_positive` to not create confusion

### CODE REVIEW

File: docs/stoploss.md [hunk 1]
```diff
@@ -3,74 +3,98 @@
...
 
-## Stop Loss support
+## Stop Loss Types
 
 At this stage the bot contains the following stoploss support modes:
 
 1. static stop loss, defined in either the strategy or configuration.
 2. trailing stop loss, defined in the configuration.
 3. trailing stop loss, custom positive loss, defined in configuration.
+4. trailing stop loss only once the trade has reached a certain offset,
```
Comment (hroff-1902) [inline docs/stoploss.md]: no trailing comma

File: docs/stoploss.md [hunk 2]
```diff
@@ -3,74 +3,98 @@
...
+This same logic will reapply a stoploss order on the exchange should you cancel it accidentally.
 
 !!! Note
     Stoploss on exchange is only supported for Binance as of now.
 
 ## Static Stop Loss
 
-This is very simple, basically you define a stop loss of x in your strategy file or alternative in the configuration, which
-will overwrite the strategy definition. This will basically try to sell your asset, the second the loss exceeds the defined loss.
+This is very simple, you define a stop loss of x. This will try to sell the asset once the loss exceeds the defined loss.
```
Comment (hroff-1902) [inline docs/stoploss.md]: maybe
"a stop loss of x (as a ratio of price, i.e. x * 100% of price)."
or some detailization like this can be handy

File: docs/stoploss.md [hunk 3]
```diff
@@ -3,74 +3,98 @@
...
 
-In case of stoploss on exchange there is another parameter called `stoploss_on_exchange_interval`. This configures the interval in seconds at which the bot will check the stoploss and update it if necessary. As an example in case of trailing stoploss if the order is on the exchange and the market is going up then the bot automatically cancels the previous stoploss order and put a new one with a stop value higher than previous one. It is clear that the bot cannot do it every 5 seconds otherwise it gets banned. So this parameter will tell the bot how often it should update the stoploss order. The default value is 60 (1 minute).
+In case of stoploss on exchange there is another parameter called `stoploss_on_exchange_interval`. This configures the interval in seconds at which the bot will check the stoploss and update it if necessary.
+
+For example, assuming the stoploss is on exchange, and trailing stoploss is enabled, and the market is going up, then the bot automatically cancels the previous stoploss order and puts a new one with a stop value higher than the previous stoploss order.
+The bot cannot do this every 5 seconds, otherwise it would get banned by the exchange.
```
Comment (hroff-1902) [inline docs/stoploss.md]: "every 5 seconds (i.e. at each throttling iteration)"

File: docs/stoploss.md [hunk 4]
```diff
@@ -3,74 +3,98 @@
...
-* which means your stop loss, gets triggered once your asset dropped below 98$
-* assuming your asset now increases to 102$
-* your stop loss, will now be 2% of 102$ or 99.96$
-* now your asset drops in value to 101$, your stop loss, will still be 99.96$
+* the bot buys an asset at a price of 100$
+* the stop loss is defined at 2%
+* the stop loss would get triggered once the asset dropps below 98$
+* assuming the asset now increases to 102$
+* the stop loss will now be 2% of 102$ or 99.96$
+* now the asset drops in value to 101$, the stop loss, will still be 99.96$, and would trigger at 99.96$.
```
Comment (hroff-1902) [inline docs/stoploss.md]: ", the stop loss will still be 99.96$ and ", no commas
